server/plugin/configuration.go

```diff
@@ -144,9 +144,10 @@ func (p *Plugin) OnConfigurationChange() error {
 	p.setConfiguration(configuration)
 
 	if oldEncryptionSecret != "" && oldEncryptionSecret != p.getConfiguration().EncryptionSecret {
-		if err := p.store.DeleteUserTokenOnEncryptionSecretChange(); err != nil {
-			p.API.LogError("Error in deleting Users.", "Error", err.Error())
-			return err
+		ch := make(chan error, 1)
+		go p.store.DeleteUserTokenOnEncryptionSecretChange(ch)
+		if len(ch) == 1 {
```

Comment:
This is not how channels work. This condition will always be true. Please read more about channels. In fact, there's no need of using channels here or logging the error here.

Reply:
Okay, I will remove it, but I think that this condition will not be always true. If we do not insert anything into the channel then its length would obviously be 0 only.